src/NUnitFramework/tests/Constraints/CollectionContainsConstraintTests.cs

```diff
@@ -51,6 +51,32 @@ public void CanTestContentsOfObjectList()
             Assert.That(list, new CollectionContainsConstraint(item));
         }
 
+        [Test]
+        public void CanTestContentsOfDictionary()
```

Comment:
This is a pretty massive test that does many things. I would prefer if it was broken down into smaller tests. It looks like your kvExists123 are good candidates for a `TestCaseSource` and your kvDoesNotExist for a separate test?